spikeinterface/core/waveform_extractor.py

```diff
@@ -257,44 +301,49 @@ def get_template(self, unit_id, mode='median', quantile_value=0.5):
             else:
-                wfs = self.get_waveforms(unit_id)
+                wfs = self.get_waveforms(unit_id, sparsity=sparsity)
                 template = np.median(wfs, axis=0)
-                self._template_median[unit_id] = template
+                if sparsity is None:
+                    self._template_median[unit_id] = template
                 return template
         elif mode == 'average':
-            if unit_id in self._template_average:
+            if unit_id in self._template_average and sparsity is None:
                 return self._template_average[unit_id]
             else:
-                wfs = self.get_waveforms(unit_id)
+                wfs = self.get_waveforms(unit_id, sparsity=sparsity)
                 template = np.average(wfs, axis=0)
-                self._template_average[unit_id] = template
+                if sparsity is None:
+                    self._template_average[unit_id] = template
                 return template
         elif mode == 'std':
-            if unit_id in self._template_std:
+            if unit_id in self._template_std and sparsity is None:
                 return self._template_std[unit_id]
             else:
-                wfs = self.get_waveforms(unit_id)
+                wfs = self.get_waveforms(unit_id, sparsity=sparsity)
                 template = np.std(wfs, axis=0)
-                self._template_std[unit_id] = template
+                if sparsity is None:
+                    self._template_std[unit_id] = template
                 return template
         elif mode == 'quantile':
-            if quantile_value in self._template_quantile:
-                if unit_id in self._template_quantile[quantile_value]:
-                    return self._template_quantile[quantile_value][unit_id]
+            if quantile_value in self._template_quantile and unit_id in self._template_quantile[quantile_value]\
+                    and sparsity is None:
+                return self._template_quantile[quantile_value][unit_id]
             else:
-                wfs = self.get_waveforms(unit_id)
+                wfs = self.get_waveforms(unit_id, sparsity=sparsity)
                 template = np.quantile(wfs, quantile_value, axis=0)
-                if quantile_value not in self._template_quantile:
-                    self._template_quantile[quantile_value] = dict()
-                self._template_quantile[quantile_value][unit_id] = template
+                if sparsity is None:
+                    if quantile_value not in self._template_quantile:
+                        self._template_quantile[quantile_value] = dict()
+                    self._template_quantile[quantile_value][unit_id] = template
                 return template
 
-    def get_all_templates(self, unit_ids=None, mode='median', quantile_value=0.5):
+    def get_all_templates(self, unit_ids=None, mode='median', quantile_value=0.5,
+                          sparsity=None):
```

Comment:
I thought we decided to not have sparsity here because it is ambiguous.
And the only use case is phy. So we could move the logic in phy.

Reply:
No we decided to have sparsity here